### ISSUE TITLE
[Parser] Autorest ignores 'properties' when '$ref' is present

### ISSUE THREAD

**Issue by jianghaolu** [Azure/autorest#2762]
In this Swagger: https://github.com/Azure/azure-rest-api-specs/blob/current/specification/web/resource-manager/Microsoft.Web/2016-08-01/WebApps.json#L15939

This is a valid Swagger and `addOnPackages` should be present in the `MSDeploy" model.

**Comment by olydis** [comment]
nope, that is actually not valid - `$ref` has to be standalone as per the Swagger spec. We only slightly "weakened" that rule and also allow simple things like `descriptions` to be siblings of `$ref`, since sometimes one can provide better documentation for the *usage* of a type compared to its *definition*. Another typical example is marking a property as `readOnly` while its schema is dragged in from somewhere else. Here, however, we have something that I wouldn't even know how to interpret! What type would that property have? One that is the composition between `MSDeployCore` and some one-property type? Or is there inheritance involved?

Long story short, this is invalid Swagger. This can easily modeled correctly, say, by using `allOf` (I assume that's what the intention was here?)

**Comment by olydis** [comment]
ah okay, it's accepted by the schema since the standard actually says that siblings of `$ref` are to be *ignored* (but can still be there). So we're behaving correct in that Swagger.

**Comment by jianghaolu** [comment]
Thanks @olydis for the explanations! Exactly what I need to forward to the service owners!
